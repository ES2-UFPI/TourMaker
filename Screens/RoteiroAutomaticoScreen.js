import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Table, TableWrapper, Row, Cell } from 'react-native-table-component';
import {Picker} from '@react-native-picker/picker';

import CustomButton from './ScreensModules/CustomButton';
import CustomMapView from "./ScreensModules/CustomMapView";
import ReactMaps from '../APIs/ReactMaps';

/*
    Usar função abaixo para acessar parametro passados de tela para tela:
        this.props.route.params
    
    Usar para passar props de uma tela a outra, coloque nos objetos:
        paramsRoteiroAutomatico
        paramsRoteiroManual
*/

class RoteiroAutomaticoScreen extends Component{

    constructor(props) {
        super(props);
        this.state = {
            tipoPDI: " ",
            PDI:" ",
            listaPDI:[],
            tiposPDI: ["Alimentação","Compras", "Hospedagem", "Entretenimento", "Outros"], //Separar a categoria dos PDIs minerados, a ser feito
            listaPDIselecao:[],
            tableHead: ["Parada", 'Excluir'],//atualizar com os dados dos PDIs mineredos
            tableData: []

        };
    }

    searchPDI(tipoPDI){
        // a fazer: procurar dentro do mapa as pdis proximos do tipo escolhido e mostrar no mapa
        var placeholderListaPDI = ["lugar mágico","lugar legal","etc"];
        this.state.listaPDIselecao= placeholderListaPDI;
    }

    tableHandle(data){
        var tableData = this.state.tableData
        tableData.push([data," "])//atualizar campos com dados dos PDIs mineredos
        this.setState({
            tableData: tableData
        })
    }

    controlRepet(data){
        var a = this.state.listaPDIselecao.filter( item => {
            return item !== data
        })
        this.setState({
            listaPDIselecao: a
        })
    }

    render(){

        console.log(this.props.route.params)
        const paramsRota = {
            listaPDI: this.state.listaPDI
        }
        
        return(
            <View style={styles.RoteiroAutomaticoScreen}>
                
                <Text>Criar Novo Roteiro Automático</Text>
                <Text>Descubra a melhor rota para os seus interesses</Text>
                <Picker
                    selectedValue={this.state.tipoPDI}
                    style={{ height: 50, width: 200 }}
                    onValueChange={(itemValue, itemIndex) =>
                        {this.setState({ tipoPDI: itemValue }),
                        this.searchPDI( this.state.tipoPDI ) }
                    }>
                    <Picker.Item label= "Tipo"/>
                    {this.state.tiposPDI.map((item, index) => {
                        return (<Picker.Item value = {item} label={item} key = {item}/>)
                    })}
                </Picker>

                <CustomMapView style = {{height: 200}}/>

                <View style={styles.container}>
                    <Table borderStyle={{ borderColor: 'transparent' }}>
                        <Row data={this.state.tableHead} style={styles.head} textStyle={styles.text} />
                        {
                            this.state.tableData.map((rowData, index) => (
                                <TableWrapper key={index} style={styles.row}>
                                    {
                                        rowData.map((cellData, cellIndex) => (
                                            <Cell key={cellIndex} data={cellIndex === 1 ? element(cellData, index) : cellData} textStyle={styles.text}/>
                                        ))
                                    }
                                </TableWrapper>
                            ))
                        }
                    </Table>
                    
                </View>
                
                <CustomButton
                    title="Criar Roteiro Automático"
                    color={styles.Buttons.color}
                    onPress={()=>{
                        this.props.navigation.navigate('Rota', paramsRota)
                    }}
                />
            </View>
        )
    }
}

const styles = StyleSheet.create({
    RoteiroAutomaticoScreen: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'flex-start',
      },
      Buttons:{
          color: '#1abc9c'
      },
      container: { flex: 1, padding: 16, paddingTop: 30, backgroundColor: '#fff' },
      head: { height: 40,width: 400, backgroundColor: '#1abc9c' },
      text: { margin: 6 },
      row: { flexDirection: 'row', backgroundColor: '#f1f8ff' },
      btn: { width: 58, height: 18, backgroundColor: '#78B7BB',  borderRadius: 2 },
      btnText: { textAlign: 'center', color: '#fff' } 
    });
export default RoteiroAutomaticoScreen;