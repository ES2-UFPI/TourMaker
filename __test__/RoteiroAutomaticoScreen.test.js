import React from 'react';
import renderer from 'react-test-renderer';

import RoteiroAutomaticoScreen from '../Screens/RoteiroAutomaticoScreen';

describe('<RoteiroAutomaticoScreen />', () => {
    it('has 1 child', () => {
        const tree = renderer.create(<RoteiroAutomaticoScreen />).toJSON();
        expect(tree.children.length).toBe(6);
    });
});

it('renders correctly', () => {
    const tree = renderer.create(<RoteiroAutomaticoScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});