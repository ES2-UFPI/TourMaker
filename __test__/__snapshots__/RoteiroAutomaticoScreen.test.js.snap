// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text>
    Criar Novo Roteiro Automático
  </Text>
  <Text>
    Descubra a melhor rota para os seus interesses
  </Text>
  <View
    style={
      Object {
        "height": 50,
        "width": 200,
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "Tipo",
            "testID": undefined,
            "textColor": undefined,
            "value": undefined,
          },
          Object {
            "label": "Alimentação",
            "testID": undefined,
            "textColor": undefined,
            "value": "Alimentação",
          },
          Object {
            "label": "Compras",
            "testID": undefined,
            "textColor": undefined,
            "value": "Compras",
          },
          Object {
            "label": "Hospedagem",
            "testID": undefined,
            "textColor": undefined,
            "value": "Hospedagem",
          },
          Object {
            "label": "Parque de Diversões",
            "testID": undefined,
            "textColor": undefined,
            "value": "Parque de Diversões",
          },
          Object {
            "label": "Galeria de Arte",
            "testID": undefined,
            "textColor": undefined,
            "value": "Galeria de Arte",
          },
          Object {
            "label": "Biblioteca",
            "testID": undefined,
            "textColor": undefined,
            "value": "Biblioteca",
          },
          Object {
            "label": "Atração Turistica",
            "testID": undefined,
            "textColor": undefined,
            "value": "Atração Turistica",
          },
          Object {
            "label": "Zoologico",
            "testID": undefined,
            "textColor": undefined,
            "value": "Zoologico",
          },
          Object {
            "label": "Museu",
            "testID": undefined,
            "textColor": undefined,
            "value": "Museu",
          },
          Object {
            "label": "Cinema",
            "testID": undefined,
            "textColor": undefined,
            "value": "Cinema",
          },
          Object {
            "label": "Spa",
            "testID": undefined,
            "textColor": undefined,
            "value": "Spa",
          },
          Object {
            "label": "Estádio",
            "testID": undefined,
            "textColor": undefined,
            "value": "Estádio",
          },
          Object {
            "label": "Parque",
            "testID": undefined,
            "textColor": undefined,
            "value": "Parque",
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "height": 200,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <AIRMap
      followsUserLocation={false}
      handlePanDrag={false}
      initialRegion={null}
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      onRegionChangeComplete={[Function]}
      onUserLocationChange={[Function]}
      region={null}
      showsUserLocation={false}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "padding": 16,
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "borderBottomWidth": 0,
            "borderColor": "transparent",
            "borderLeftWidth": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#1abc9d",
              "width": 400,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Tipo
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Quantidade
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Excluir
          </Text>
        </View>
      </View>
      <RCTScrollView>
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#1abc9c",
                },
              ]
            }
          >
            Criar Roteiro Automático
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
